node/pkg/node/node_test.go

```diff
@@ -213,6 +233,51 @@ func waitForHeartbeats(t *testing.T, zapObserver *observer.ObservedLogs, gs []*m
 	}
 }
 
+func waitForHeartbeatsInMetrics(t *testing.T, ctx context.Context, gs []*mockGuardian) {
+	requests := make([]*http.Request, len(gs))
+	//logger := supervisor.Logger(ctx)
+
+	// create the prom api clients
+	for i := range gs {
+		url := fmt.Sprintf("http://localhost:%d/metrics", mockStatusPort(uint(i)))
+		req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+		assert.NoError(t, err)
+		requests[i] = req
+	}
+
+	// query them
+	for readyCounter := 0; readyCounter < len(gs); {
+		for i, g := range gs {
+			if g.ready {
+				continue
+			}
+
+			resp, err := http.DefaultClient.Do(requests[i])
+			if err != nil {
+				resp.Body.Close()
+				continue
+			}
```

Comment:
What closes the response body here if there is not an error? The general pattern is to defer a resp.Body.Close(), but this is looped so does it make sense to do it at the end or just let them get torn down by the test exiting?